### DISCUSSION TITLE
Error in Virtual Device Contexts v4.0.1

### DISCUSSION THREAD

**Discussion by jlhuston** [netbox-community/netbox#16086] [Help Wanted!] [answered]
When attempting to view the whole list of Virtual Device Contexts, I get a 500 error page with the below details. When I click on an existing VDC used on a device to get to the detail page (/dcim/virtual-device-contexts/##), that works just fine. It is just the root page for VDCs that is having the issue.

FieldError at /dcim/virtual-device-contexts/
Cannot resolve keyword '_name' into field. Choices are: bookmarks, comments, created, custom_field_data, description, device, device_id, id, identifier, interface_count, interfaces, journal_entries, last_updated, name, primary_ip4, primary_ip4_id, primary_ip6, primary_ip6_id, status, tagged_items, tags, tenant, tenant_id
Request Method:	GET
Request URL:	https://netbox.gates.ventures/dcim/virtual-device-contexts/
Django Version:	5.0.6
Exception Type:	FieldError
Exception Value:	
Cannot resolve keyword '_name' into field. Choices are: bookmarks, comments, created, custom_field_data, description, device, device_id, id, identifier, interface_count, interfaces, journal_entries, last_updated, name, primary_ip4, primary_ip4_id, primary_ip6, primary_ip6_id, status, tagged_items, tags, tenant, tenant_id
Exception Location:	/opt/netbox-4.0.1/venv/lib/python3.10/site-packages/django/db/models/sql/query.py, line 1772, in names_to_path
Raised during:	dcim.views.VirtualDeviceContextListView
Python Executable:	/opt/netbox-4.0.1/venv/bin/python3.10
Python Version:	3.10.12
Python Path:	
['/opt/netbox/netbox',
 '/opt/netbox-4.0.1',
 '/opt/netbox-4.0.1/venv/bin',
 '/usr/lib/python310.zip',
 '/usr/lib/python3.10',
 '/usr/lib/python3.10/lib-dynload',
 '/opt/netbox-4.0.1/venv/lib/python3.10/site-packages']
Server time:	Fri, 10 May 2024 22:17:22 +0000

Traceback data is:
```
Environment:


Request Method: GET
Request URL: https://netbox.gates.ventures/dcim/virtual-device-contexts/

Django Version: 5.0.6
Python Version: 3.10.12
Installed Applications:
['django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.humanize',
 'django.forms',
 'corsheaders',
 'debug_toolbar',
 'django_filters',
 'django_htmx',
 'django_tables2',
 'django_prometheus',
 'strawberry_django',
 'mptt',
 'rest_framework',
 'social_django',
 'taggit',
 'timezone_field',
 'core',
 'account',
 'circuits',
 'dcim',
 'ipam',
 'extras',
 'tenancy',
 'users',
 'utilities',
 'virtualization',
 'vpn',
 'wireless',
 'django_rq',
 'drf_spectacular',
 'drf_spectacular_sidecar']
Installed Middleware:
['strawberry_django.middlewares.debug_toolbar.DebugToolbarMiddleware',
 'corsheaders.middleware.CorsMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'django_htmx.middleware.HtmxMiddleware',
 'netbox.middleware.RemoteUserMiddleware',
 'netbox.middleware.CoreMiddleware',
 'netbox.middleware.MaintenanceModeMiddleware']



Traceback (most recent call last):
  File "/opt/netbox-4.0.1/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/opt/netbox-4.0.1/venv/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/netbox-4.0.1/venv/lib/python3.10/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/opt/netbox/netbox/netbox/views/generic/base.py", line 77, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/opt/netbox/netbox/utilities/views.py", line 108, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/opt/netbox-4.0.1/venv/lib/python3.10/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
  File "/opt/netbox/netbox/netbox/views/generic/bulk_views.py", line 162, in get
    table = self.get_table(self.queryset, request, has_bulk_actions)
  File "/opt/netbox/netbox/netbox/views/generic/mixins.py", line 53, in get_table
    table.configure(request)
  File "/opt/netbox/netbox/netbox/tables/tables.py", line 162, in configure
    self.order_by = ordering
  File "/opt/netbox-4.0.1/venv/lib/python3.10/site-packages/django_tables2/tables.py", line 542, in order_by
    self.data.order_by(self._order_by)
  File "/opt/netbox-4.0.1/venv/lib/python3.10/site-packages/django_tables2/data.py", line 224, in order_by
    self.data = self.data.order_by(*order_by_accessors)
  File "/opt/netbox-4.0.1/venv/lib/python3.10/site-packages/django/db/models/query.py", line 1701, in order_by
    obj.query.add_ordering(*field_names)
  File "/opt/netbox-4.0.1/venv/lib/python3.10/site-packages/django/db/models/sql/query.py", line 2253, in add_ordering
    self.names_to_path(item.split(LOOKUP_SEP), self.model._meta)
  File "/opt/netbox-4.0.1/venv/lib/python3.10/site-packages/django/db/models/sql/query.py", line 1772, in names_to_path
    raise FieldError(

Exception Type: FieldError at /dcim/virtual-device-contexts/
Exception Value: Cannot resolve keyword '_name' into field. Choices are: bookmarks, comments, created, custom_field_data, description, device, device_id, id, identifier, interface_count, interfaces, journal_entries, last_updated, name, primary_ip4, primary_ip4_id, primary_ip6, primary_ip6_id, status, tagged_items, tags, tenant, tenant_id
```

**Comment by jlhuston** [comment]
I did take a quick look at the dcim_virtualdevicecontext table in the database and didn't find any rows that referenced device IDs that don't exist as actual devices. Just in case it was some sort of data inconsistency.

**Comment by CooperCOK** [comment]
I am facing the exact issue. still existst in netbox 4.0.6. Any help would be greatly appriciated

**Comment by jlhuston** [comment]
Still there in 4.4.1. Not sure what's going on or how to resolve? Are there troubleshooting data that would be useful to anyone to know?

**Reply by pheus** [reply, marked as answer]
Did you try clearing your table preferences for Virtual Device Contexts?

Open doc issue: https://github.com/netbox-community/netbox/issues/20449

**Reply by jlhuston** [reply]
You are correct; that fixed the issue. I only had to clear the table configuration for the virtual device contexts to restore functionality.

This also blocked anything that showed a rack elevation SVG (Elevations or a Device page that showed the SVG) in addition to the Virtual Device Contexts table. So, that table preference configuration is a bit more far reaching than one would anticipate under normal circumstances.

Again - clearing the preferences as per issue 20449 resolved the problem for me.

Thank you!
